ui/static/ui/js/listing.js

```diff
@@ -126,7 +126,57 @@ define('listing',
         }
       });
 
+      /**
+       * Render resource items in the UI
+       *
+       * @returns {ReactElement} The rendered resource items
+       */
       var renderListingResources;
+      /**
+       * Render pagination in the UI
+       *
+       * @returns {ReactElement} The rendered pagination
+       */
+      var renderPagination;
+
+      // queryMap is the canonical place to manage query parameters for the UI.
+      // The URL in the browser will be updated with these changes in
+      // refreshFromAPI.
+      var queryMap = {};
+      // Populate queryMap with query string key value pairs.
+      _.each(URI(window.location).query(true), function(v, k) {
+        if (!Array.isArray(v)) {
+          // URI().query(true) will put two or more values with the same
+          // key in an array and not use an array for single values.
+          // Put everything into arrays for consistency's sake.
+          queryMap[k] = [v];
+        } else {
+          queryMap[k] = v;
+        }
+      });
+
+      /**
+       * Iterates through query string to figure out which facets
+       * are checked, and return an object with this information.
+       */
```

Comment:
this looks hacky: maybe we should make this in the backend and return a "tokenized" query parameter object